api.go

```diff
@@ -403,23 +403,18 @@ type CloudStorageClient interface {
 	SetBucketACL(bucketName string, cannedACL BucketACL) error
 	GetBucketACL(bucketName string) (BucketACL, error)
 
-	ListBuckets() ([]BucketStat, error)
-	ListObjects(bucket, prefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectStat
-	ListIncompleteUploads(bucket, prefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectMultipartStat
+	ListBuckets() ([]BucketInfo, error)
+	ListObjects(bucket, prefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectInfo
+	ListIncompleteUploads(bucket, prefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectMultipartInfo
 
 	// Object Read/Write/Stat operations.
-	GetObject(bucketName, objectName string) (reader io.ReadCloser, stat ObjectStat, err error)
-	PutObject(bucketName, objectName string, data io.Reader, size int64, contentType string) (n int64, err error)
-	StatObject(bucketName, objectName string) (ObjectStat, error)
+	GetObject(bucketName, objectName string) (reader *Object, err error)
+	PutObject(bucketName, objectName string, data io.Reader, contentType string) (n int64, err error)
+	StatObject(bucketName, objectName string) (ObjectInfo, error)
 	RemoveObject(bucketName, objectName string) error
 	RemoveIncompleteUpload(bucketName, objectName string) error
 
-	// Object Read/Write for sparse upload.
-	GetObjectPartial(bucketName, objectName string) (reader ReadAtCloser, stat ObjectStat, err error)
-	PutObjectPartial(bucketName, objectName string, data ReadAtCloser, size int64, contentType string) (n int64, err error)
-
 	// File to Object API.
-	FPutObject(bucketName, objectName, filePath, contentType string) (n int64, err error)
 	FGetObject(bucketName, objectName, filePath string) error
```

Comment:
This makes me think. Since we cannot let FGetObject go, we can keep FPutObject as well. However, GetObject and PutObject changes remain as is (using reader type).